pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Topics.java

```diff
@@ -1636,6 +1636,27 @@ void expireMessagesForAllSubscriptions(String topic, long expireTimeInSeconds)
      */
     CompletableFuture<Void> expireMessagesForAllSubscriptionsAsync(String topic, long expireTimeInSeconds);
 
+    /**
+     * Peek messages from a topic subscription.
+     *
+     * @param topic
+     *            topic name
+     * @param subName
+     *            Subscription name
+     * @param  offset
+     *            Start index to start reading messages.
+     * @param numMessages
+     *            Number of messages
+     * @return
+     * @throws NotAuthorizedException
+     *             Don't have admin permission
+     * @throws NotFoundException
+     *             Topic or subscription does not exist
+     * @throws PulsarAdminException
+     *             Unexpected error
+     */
+    List<Message<byte[]>> peekMessages(String topic, String subName, int offset, int numMessages) throws PulsarAdminException;
```

Comment:
Pulsar does not use the "offset" concept like Kafka or some other messaging systems. So it might be a little confusing here. Instead, you can use `messagePosition` like the name in https://github.com/apache/pulsar/blob/c36c18d44fd4df07bce9b7961c59685e0e91a420/pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/internal/TopicsImpl.java#L870

Reply:
BTW, after introducing these new methods, you can change the original calls to default methods and then you won't need to copy the API docs to this new method.

```java
    /**
     * <original API docs...>
     * @param  offset
     *            Start index to start reading messages.
     * <original API docs...>
     */
    List<Message<byte[]>> peekMessages(String topic, String subName, int startPosition, int numMessages) throws PulsarAdminException;

    default List<Message<byte[]>> peekMessages(String topic, String subName, int numMessages) throws PulsarAdminException {
        return peekMessages(topic, subName, 1, numMessages);
    }
```

Reply:
Thanks for looking into this. The idea of introducing offset as one of the method parameters was to be overwrite offset value of 1 and pass in any number as a startingIndex to fetch numMessages of sorts.   This would potentially help in paging while pulling messages from a subscription backlog.